src/blockchain/src/block.rs

```diff
@@ -48,11 +48,12 @@ pub fn get_publickey_from_keypair(
     keypair.public()
 }
 
-// struct Block define a block strcuture
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
 pub struct Block {
     pub header: Header,
+    /// `transactions` The transactions that make up this block
     pub transactions: Vec<Transaction>,
+    /// `signature` a signature of the whole block - it's header and transactions
     pub signature: BlockSignature,
 }
```

Comment:
You dont need to duplicate the field name

```suggestion
    pub header: Header,
    /// The list of transactions that make up a block
    pub transactions: Vec<Transaction>,
    /// A signature of the entire block - it's header and transactions
    pub signature: BlockSignature,
}
```